File: app_mobile/kmp/androidApp/src/main/java/com/ansmi/gestsquadre/kmp/map/SquadIcons.kt
package com.ansmi.gestsquadre.kmp.map

import androidx.annotation.DrawableRes
import com.ansmi.gestsquadre.kmp.R

object SquadIcons {
    @DrawableRes
    fun drawableRes(iconKey: String?): Int =
        when (iconKey?.trim().orEmpty()) {
            "ambulanza" -> R.drawable.squad_ambulanza
            "coordinatore_cri" -> R.drawable.squad_coordinatore_cri
            "vigili_fuoco" -> R.drawable.squad_vigili_fuoco
            "forze_ordine" -> R.drawable.squad_forze_ordine
            "medico" -> R.drawable.squad_medico
            "fig" -> R.drawable.squad_fig
            "squadre_a_piedi", "" -> R.drawable.squad_squadre_a_piedi
            else -> R.drawable.squad_squadre_a_piedi
        }
}

File: app_mobile/kmp/iosApp/iosApp/SquadIcons.swift
import UIKit

/// Icone squadra su mappa (PNG nel bundle). Allineato ad Android SquadIcons.kt.
enum SquadIcons {
    static func normalizeKey(_ raw: String?) -> String {
        let key = raw?.trimmingCharacters(in: .whitespacesAndNewlines).lowercased() ?? ""
        if key.isEmpty { return "squadre_a_piedi" }
        switch key {
        case "nv_ansmi", "ansmi", "k9_nvansmi": return "logo_ansmi"
        case "rolexpress", "rolex_express", "role_xpress": return "logo_rolexpress"
        case "waypoint_toc", "toc", "t.o.c.": return "waypoint_toc"
        default:
            return key.replacingOccurrences(of: "-", with: "_")
        }
    }

    static func bundleImageName(forKey iconKey: String?) -> String {
        let key = normalizeKey(iconKey)
        switch key {
        case "ambulanza": return "squad_ambulanza"
        case "coordinatore_cri": return "squad_coordinatore_cri"
        case "vigili_fuoco": return "squad_vigili_fuoco"
        case "forze_ordine": return "squad_forze_ordine"
        case "medico": return "squad_medico"
        case "fig": return "squad_fig"
        case "waypoint_toc": return "squad_waypoint_toc"
        case "logo_rolexpress": return "squad_logo_rolexpress"
        case "logo_ansmi": return "squad_logo_ansmi"
        case "squadre_a_piedi": return "squad_squadre_a_piedi"
        default:
            let candidate = "squad_\(key)"
            return UIImage(named: candidate) != nil ? candidate : "squad_squadre_a_piedi"
        }
    }
}
